### PR TITLE
Make it faster, show me what your mama gave you

### PR DESCRIPTION
**Pull request by vertexclique**

### CODE REVIEW

File: artillery-core/src/epidemic/membership.rs [hunk 1]
```diff
@@ -148,7 +148,7 @@ impl ArtilleryMemberList {
                                 .clone();
                         let new_host = new_member
                             .remote_host()
-                            .or(entry.get().remote_host())
+                            .or_else(|| entry.get().remote_host())
```
Comment (o0Ignition0o) [inline artillery-core/src/epidemic/membership.rs:151]: lazy eval ftw !

File: artillery-core/src/epidemic/state.rs [hunk 2]
```diff
@@ -402,44 +400,41 @@ impl ArtilleryState {
...
-        sender: sender.clone(),
-        cluster_key: cluster_key.clone(),
+        sender: *sender,
+        cluster_key: cluster_key.into(),
         request: request.clone(),
         state_changes: Vec::new(),
     };
 
-    for i in 0..state_changes.len() + 1 {
+    for i in 0..=state_changes.len() {
```
Comment (o0Ignition0o) [inline artillery-core/src/epidemic/state.rs:432]: nice one !

**Review by o0Ignition0o — APPROVED** [review]
LGTM :shipit:

**Comment by vertexclique** [comment]
🚀 going in! Thanks @o0Ignition0o !

**Pull request merged**